internal/slicesext/uniq.go

```diff
@@ -0,0 +1,28 @@
+package slicesext
+
+// Uniq returns a new slice with all duplicates removed.
+func Uniq[T comparable](list []T) []T {
```

Comment:
can uses slices.Compact + slices.Sort - if sorting isn't a problem

Reply:
That wouldn't work here, because:

1. I use this for a list of structs, so it isn't naturally sortable;
2. I could in theory sort by a given field, but I actually want to keep the same order, so sorting isn't ideal here.